.github/workflows/release.yaml

```diff
@@ -19,3 +18,6 @@ jobs:
       tox-version: "<4"
       snapcraft: true
       commands: "['make functional']"
+      # Select latest track for the default branch, otherwise use versioned branch name as the track name
+      track: ${{ contains(fromJSON('["main", "master"]'), github.ref_name) && 'latest' || github.ref_name }}
```

Comment:
Can you provide examples how this will work? I do not understand when `&&` and `||` will be used.

Reply:
Sure. This expression means:
```
if contains(fromJSON('["main", "master"]'), github.ref_name):
    track = "latest"
else
    track = github.ref_name
```
This is similar to the bash syntax ([example](https://stackoverflow.com/a/3953712))

To see the workflow in action:
- on the main branch: https://github.com/agileshaw/prometheus-juju-exporter/actions/runs/5017866317/jobs/8997047485 (you can check the inputs of the "Publish Snap" job are correctly set)
- on a versioned branch: https://github.com/agileshaw/prometheus-juju-exporter/actions/runs/5016042267/jobs/8993083887 (the workflow failed because there's no 2.6 track in snap store, but the inputs are set correctly)

Reply:
Alternatively, if this one-liner expression is not very readable, I could create a separate job that run a bash script to determine the track and set it to $GITHUB_OUTPUT to use in the input here.

Reply:
Maybe you can provide comment there with explanation, but it's not mandatory since this is more my ignorance to this kind of knowledge. Thanks for explanation.